pyfr/plugins/kernels/turbulence.mako

```diff
@@ -0,0 +1,65 @@
+<% import numpy as np %>
+
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%pyfr:macro name='turbulence' params='t, u, ploc, src'>
+  fpdtype_t tbc = 5.9604644775390625e-8;
+  fpdtype_t arg;
+  fpdtype_t clip;
+  fpdtype_t g;
+  fpdtype_t tpos[3];
+  fpdtype_t tploc[3];
+  fpdtype_t delta2[3];
+  fpdtype_t yzmin[2] = {${ymin}, ${zmin}};
+  fpdtype_t yzmax[2] = {${ymax}, ${zmax}};
+  fpdtype_t utilde[3] = {};
+
+  unsigned int oldstate;
+  unsigned int newstate;
+  unsigned int rshift;
+  
+  % for i, r in enumerate(rot):
+    tploc[${i}] = ${' + '.join(f'{r[j]}*(ploc[{j}] - {shift[j]})' for j in range(3))};
```

Comment:
enumerate(zip(r, shift))

Reply:
Do you mean like this?

% for i, rotc in enumerate(rot):
    tploc[${i}] = ${' + '.join(f'{r}*(ploc[{j}] - {s})' for j, (r,s) in enumerate(zip(rotc, shift)))};
  % endfor

Reply:
Yes, although you may be able to use pyfr.dot to clean things up a bit further.